### PR TITLE
[Autocomplete] Deprecate `ExtraLazyChoiceLoader` in favor of `LazyChoiceLoader`

### PR DESCRIPTION
**Pull request by norkunas**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| Issues        | N/A
| License       | MIT

### CODE REVIEW

**Comment by norkunas** [comment]
fabbot CS suggestion/warning does not make sense

**Review by yceruto — APPROVED** [review]

File: src/Autocomplete/src/Form/ChoiceList/Loader/ExtraLazyChoiceLoader.php [hunk 1]
```diff
@@ -17,6 +17,8 @@
 
 /**
  * Loads choices on demand only.
+ *
+ * @deprecated since Autocomplete 2.23, use `Symfony\Component\Form\ChoiceList\Loader\LazyChoiceLoader` instead.
```
Comment (smnandre) [inline src/Autocomplete/src/Form/ChoiceList/Loader/ExtraLazyChoiceLoader.php]: Should this be in the constructor or something ? Or this docblock will trigger a warning ? (genuine question)

Comment (smnandre) [inline src/Autocomplete/src/Form/ChoiceList/Loader/ExtraLazyChoiceLoader.php]: « and will be removed in 3.0 » ?

**Review by smnandre — COMMENTED** [review]
👍  great idea

File: src/Autocomplete/src/Form/ChoiceList/Loader/ExtraLazyChoiceLoader.php (hunk 1 above)
Comment (norkunas) [reply src/Autocomplete/src/Form/ChoiceList/Loader/ExtraLazyChoiceLoader.php]: > Should this be in the constructor or something

I don't know :)

> Or this docblock will trigger a warning ?

in Symfony, yes :)

Comment (norkunas) [reply src/Autocomplete/src/Form/ChoiceList/Loader/ExtraLazyChoiceLoader.php]: updated

**Review by smnandre — APPROVED** [review]
Thank you @norkunas!

**Comment by smnandre** [comment]
Thank you Tomas.

**Pull request merged**